tsl/src/bgw_policy/policies_v2.c

```diff
@@ -158,13 +158,17 @@ validate_and_create_policies(policies_info all_policies, bool if_exists)
 	if (all_policies.refresh && !IS_INTEGER_TYPE(all_policies.partition_type))
 	{
 		/* Check if there are any gaps in the refresh policy */
-		refresh_window_size = (start_offset == ts_time_get_max(all_policies.partition_type) ||
-							   end_offset == ts_time_get_min(all_policies.partition_type)) ?
-								  start_offset :
-								  start_offset - end_offset;
+		uint64 refresh_window_size =
+			(start_offset == ts_time_get_max(all_policies.partition_type) ||
+			 end_offset == ts_time_get_min(all_policies.partition_type)) ?
+				start_offset :
+				start_offset - end_offset;
```

Comment:
Since `start_offset` is signed and `end_offset` is also signed, you run the risk of getting warning that this result can be negative when cast to unsigned. You can probably solve it by adding an assert before this line.